src/components/Accordion/Accordion.test.tsx

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+import userEvent from '@testing-library/user-event';
+import { render as renderRtl, screen } from '@testing-library/react';
+
+import type { AccordionProps } from './Accordion';
+import { Accordion } from './Accordion';
+import { AccordionHeader } from './AccordionHeader';
+import { AccordionContent } from './AccordionContent';
+
+const render = (props: Partial<AccordionProps> = {}) => {
+  const allProps = {
+    children: (
+      <>
+        <AccordionHeader>AccordionHeader</AccordionHeader>
+        <AccordionContent>AccordionContent</AccordionContent>
+      </>
+    ),
+    onClick: jest.fn(),
+    open: true,
+    ...props,
+  };
+  renderRtl(<Accordion {...allProps} />);
+};
+
+const user = userEvent.setup();
+
+describe('Accordion', () => {
+  it('should call handleClick with open=true when AccordionHeader is clicked', async () => {
+    const handleClick = jest.fn();
+    render({ onClick: handleClick, open: true });
+
+    await user.click(screen.getByRole('button', { name: 'AccordionHeader' }));
+    expect(handleClick).toHaveBeenCalledTimes(1);
+  });
+});
+
+describe('Accordion', () => {
+  it('should call handleClick with open=false when AccordionHeader is clicked', async () => {
+    const handleClick = jest.fn();
+    render({ onClick: handleClick, open: false });
+
+    await user.click(screen.getByRole('button', { name: 'AccordionHeader' }));
+    expect(handleClick).toHaveBeenCalledTimes(1);
+  });
+});
+
+describe('Accordion', () => {
+  it('should have aria-expanded=true when clicked and open=true', async () => {
+    const handleClick = jest.fn();
+    render({ onClick: handleClick });
+
+    await user.click(screen.getByRole('button', { name: 'AccordionHeader' }));
+    expect(
+      screen.getByRole('button', { name: 'AccordionHeader', expanded: true }),
+    ).toBeInTheDocument();
+  });
```

Comment:
This test doesnt really test what it says - the `click` has no effect here, since the component is controlled, and we are not updating the `open`-prop. So `open` is always true, no matter how many times the button is clicked.

```suggestion
  it('should have aria-expanded=true when open=true', () => {
    render({ open: true }) // could potentially omit this, as open is default in the render helper, but good to be explicit.

    expect(
      screen.getByRole('button', { name: 'AccordionHeader', expanded: true }),
    ).toBeInTheDocument();
  });
```

There is a similar problem for the other tests as well, which checks this on keypress etc. They should instead just verify that the `onClick` callback is fired.